lib/fhir/core.js

```diff
@@ -321,6 +324,133 @@ module.exports = (mongo, modules) => {
       })
     },
 
+    /**
+     * History for a resource. The 'query' field must be specified in the RequestContext parameter.
+     *
+     * @param {RequestContext} ctx
+     * @param {String} resourceType The type of FHIR resource
+     * @param {CoreCallback} callback
+     */
+    history: (ctx, resourceType, callback) => {
+      authorization.authorize('history', ctx, resourceType, (err, badRequest) => {
+        if (err || badRequest) {
+          return handleErrorAndBadRequest(err, badRequest, callback)
+        }
+
+        const fhirModule = modules[resourceType]
+        hooks.executeBeforeHooks('history', fhirModule, ctx, resourceType, (err, badRequest) => {
+          if (err || badRequest) {
+            return handleErrorAndBadRequest(err, badRequest, callback)
+          }
+
+          mongo.getDB((err, db) => {
+            if (err) {
+              return callback(err)
+            }
+
+            /* This doesn't work for deletes but saving in case needed later.
+            let query = [ { $lookup : { from : resourceType+'_history', localField: "id", foreignField: "id", as: 'history' } }, 
+              { $lookup : { from: resourceType, localField: 'id', foreignField: 'id', as : 'top' } }, 
+              { $project : { allHist :{ $setUnion : ["$top","$history"] } } }, 
+              { $unwind : "$allHist" }, 
+              {$replaceRoot: {newRoot:"$allHist"}} ]
+            */
+            let query = [
+              { $limit: 1 },
+              { $lookup: { from: resourceType, pipeline:[], as: "top" } }, 
+              { $lookup: { from: resourceType+"_history", pipeline: [], as: "history" } },
+              { $project: { all: { $setUnion: [ "$top", "$history" ] } } },
+              { $unwind: "$all" }, 
+              { $replaceRoot: { newRoot: "$all" } }
```

Comment:
Perhaps another way to do this would be to just to two separate queries and combine the results? Might be more efficient. Although this way seems fine to until we actually see a performance issue.